Makefile

```diff
@@ -0,0 +1,115 @@
+# This file is part of REANA.
+# Copyright (C) 2019 CERN.
+#
+# REANA is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+# configuration options that may be passed as environment variables:
+GITHUB_USER ?= anonymous
+MINIKUBE_DRIVER ?= kvm2
+MINIKUBE_PROFILE ?= minikube
+TIME_SLEEP ?= 40
+VENV_NAME ?= reana
+
+# let's detect where we are and whether minikube and kubectl are available:
+HAS_KUBECTL := $(shell command -v kubectl 2> /dev/null)
+HAS_MINIKUBE := $(shell command -v minikube 2> /dev/null)
+PWD := $(shell pwd)
+
+all: help
+
+help:
+	@echo 'Description:'
+	@echo
+	@echo '  This Makefile facilitates building and testing REANA on a local Minikube cluster.'
+	@echo '  Useful for personal development and CI testing scenarios.'
+	@echo
+	@echo 'Available commands:'
+	@echo
+	@grep -E '^[a-zA-Z_-]+:.*?# .*$$' $(MAKEFILE_LIST) | sort | awk 'BEGIN {FS = ":.*?# "}; {printf "  \033[36m%-17s\033[0m %s\n", $$1, $$2}'
+	@echo
+	@echo 'Configuration options:'
+	@echo
+	@echo '  GITHUB_USER       Which GitHub user account to use for cloning? [default=anonymous]'
+	@echo '  MINIKUBE_DRIVER   Which Minikube driver to use? [default=kvm2]'
+	@echo '  MINIKUBE_PROFILE  Which Minikube profile to use? [default=minikube]'
+	@echo '  TIME_SLEEP        How much time to sleep when bringing cluster up and down? [default=40]'
+	@echo '  VENV_NAME         Which Python virtual environment name to use? [default=reana]'
+	@echo
+	@echo 'Examples:'
+	@echo
+	@echo '  # how to set up personal development environment:'
+	@echo '  $$ GITHUB_USER=johndoe MINIKUBE_DRIVER=virtualbox make setup clone'
+	@echo '  # how to build latest checked-out sources and run an example:'
+	@echo '  $$ make ci'
+	@echo
+	@echo '  # how to perform an independent automated CI test run:'
+	@echo '  $$ mkdir /tmp/nightlybuild && cd /tmp/nightlybuild'
+	@echo '  $$ git clone https://github.com/reanahub/reana && cd reana'
+	@echo '  $$ VENV_NAME=nightlybuild MINIKUBE_PROFILE=nightlybuild make ci'
+	@echo '  $$ VENV_NAME=nightlybuild MINIKUBE_PROFILE=nightlybuild make teardown'
+	@echo '  $$ cd /tmp && rm -rf /tmp/nightlybuild'
+
+setup: # Prepare local host virtual environment and Minikube for REANA building and deployment.
+ifndef HAS_KUBECTL
+	$(error "Please install Kubectl v1.13.3 or higher")
+endif
+ifndef HAS_MINIKUBE
+	$(error "Please install Minikube v0.34 or higher")
+endif
+	minikube status --profile ${MINIKUBE_PROFILE} || minikube start --profile ${MINIKUBE_PROFILE} --vm-driver ${MINIKUBE_DRIVER} --feature-gates="TTLAfterFinished=true"
+	test -e ${HOME}/.virtualenvs/${VENV_NAME}/bin/activate || virtualenv ${HOME}/.virtualenvs/${VENV_NAME}
+
+clone: # Clone REANA source code repositories locally.
+	source ${HOME}/.virtualenvs/${VENV_NAME}/bin/activate && \
```

Comment:
I'm using pyenv so the virtualenv directory is not at this location. Not a problem but could be mentioned that it's meant to be used with virtualenv in the docs.

Reply:
Yeah, I've used vanilla `virtualenv` as per harmony with the other docs, since we use it as an example everywhere...